tokio/src/runtime/mod.rs

```diff
@@ -577,4 +581,16 @@ impl Runtime {
     pub fn shutdown_background(self) {
         self.shutdown_timeout(Duration::from_nanos(0))
     }
+
+    /// Craps out a RawFd for *nix systems, if one exists.
+    #[cfg(unix)]
+    pub fn driver_fd(&self) -> Option<std::os::unix::io::RawFd> {
```

Comment:
Would it be worthwhile to add to the commit message the rationale for exposing the driver FD or is it self-explanatory?  As the message at present talks about the one half of the even loop integration, but we could add something to say that we expose the driver FD so that tokio can express readiness to whatever event loop it is being integrated with.  Again, no need if this is seems obvious already.

Reply:
Good point-- the `RawFd` shenanigans comprise over half of the patch, so it at least deserves a note. The rightish way is probably to separate it into its own commit; probably not too painful to do.

Reply:
Split FD stuff into c5cef998e030842228da9c93689ab80863e718fd, makes for easier reading I think 👍